zeroex/orderwatch/order_watcher.go

```diff
@@ -977,32 +1229,37 @@ func (w *Watcher) findOrder(orderHash common.Hash) *types.OrderWithMetadata {
 // findOrdersByTokenAddressAndTokenID finds and returns all orders that have
 // either a makerAsset or a makerFeeAsset matching the given tokenAddress and
 // tokenID.
-func (w *Watcher) findOrdersByTokenAddressAndTokenID(makerAddress, tokenAddress common.Address, tokenID *big.Int) ([]*types.OrderWithMetadata, error) {
-	ordersWithAffectedMakerAsset, err := w.db.FindOrders(&db.OrderQuery{
-		Filters: []db.OrderFilter{
-			{
-				Field: db.OFMakerAddress,
-				Kind:  db.Equal,
-				Value: makerAddress,
-			},
-			db.MakerAssetIncludesTokenAddressAndTokenID(tokenAddress, tokenID),
+func (w *Watcher) findOrdersByTokenAddressAndTokenID(makerAddress, tokenAddress common.Address, tokenID *big.Int, filter db.OrderFilter) ([]*types.OrderWithMetadata, error) {
```

Comment:
I don't want to block this PR but at this point we should re-examine whether the `findOrdersByX` methods really make sense. These methods were written with the old database implementation in mind and it might be cleaner now to just pass around a slice of filters, adding to it as necessary.